### PR TITLE
ER-1361 Revoke Permission Employer.Web

### PR DESCRIPTION
**Pull request by chrisjensenuk**
- Now displaying the filter on the Employer Dashboard
- Added alert dismissal logic per user (Command and CommandHandler)
- Added Transferred Vacancies alert to Employer dashboard
- Added message to vacancy Manage and Edit screens

### CODE REVIEW

File: src/Employer/Employer.Web/Controllers/DashboardController.cs [hunk 1]
```diff
@@ -39,5 +42,16 @@ public async Task<IActionResult> Dashboard([FromRoute]string employerAccountId,[
 
             return View(vm);
         }
+
+        [HttpPost("dismiss-alert", Name = RouteNames.Dashboard_DismissAlert)]
+        public async Task<IActionResult> DismissAlert([FromRoute] string employerAccountId, AlertDismissalEditModel model)
+        {
+            if (Enum.TryParse(typeof(AlertType), model.AlertType, out var alertTypeEnum))
```
Comment (dev4valtech) [inline src/Employer/Employer.Web/Controllers/DashboardController.cs]: `Enum.TryParse<AlertType>(model.AlertType, out var result);`

File: src/Employer/Employer.Web/Controllers/DashboardController.cs [hunk 2]
```diff
@@ -39,5 +42,16 @@ public async Task<IActionResult> Dashboard([FromRoute]string employerAccountId,[
...
+            {
+                await _orchestrator.DismissAlert(User.ToVacancyUser(), (AlertType)alertTypeEnum);
```
Comment (dev4valtech) [inline src/Employer/Employer.Web/Controllers/DashboardController.cs]: If you use generic TryParse as suggested above, there is no need to cast the variable here.

File: src/Employer/Employer.Web/Orchestrators/DashboardOrchestrator.cs [hunk 3]
```diff
@@ -151,5 +169,25 @@ private string GetFilterHeading(int totalVacancies, FilteringOptions filteringOp
                     return $"{totalVacancies} {filterText} {"vacancy".ToQuantity(totalVacancies, ShowQuantityAs.None)}";
             }
         }
+
+        private TransferredVacanciesAlertViewModel GetTransferredVacanciesAlertViewModel(IEnumerable<VacancySummary> vacancies, DateTime? userLastDismissedDate)
```
Comment (dev4valtech) [inline src/Employer/Employer.Web/Orchestrators/DashboardOrchestrator.cs]: I was thinking ahead a little bit, there are about four types of alerts that we have anticipated. Working out all the different alerts in orchestrator may make this class too big and alerts feels like cross concern to me. I was thinking inline with having a UserAlertManager with methods Get and Dismiss. Perhaps having a single partial view that can be reused when displaying alerts on top of views other then Dashboard. 

This is just an idea, I have not completely thought this through.

File: src/Employer/Employer.Web/Views/VacancyPreview/VacancyPreview.cshtml [hunk 4]
```diff
@@ -36,9 +36,16 @@
 }
 <div class="govuk-grid-row">
     <div class="govuk-grid-column-full">
-        <partial name="@PartialNames.ValidationSummary" model='new ValidationSummaryViewModel { ModelState = ViewData.ModelState, OrderedFieldNames = Model.OrderedFieldNames }' />
+        <div asp-show="@Model.IsTransferred">
```
Comment (dev4valtech) [inline src/Employer/Employer.Web/Views/VacancyPreview/VacancyPreview.cshtml]: So this bit of view is being repeated in ManageVacancy.cshtml. This is where a partial view with any alerts could help... again still thinking out loud on my previously mentioned idea.

File: src/Shared/Recruit.Vacancies.Client/Infrastructure/QueryStore/Projections/Employer/EmployerDashboard.cs [hunk 5]
```diff
@@ -1,4 +1,5 @@
-﻿using System.Collections.Generic;
+﻿using System;
```
Comment (dev4valtech) [inline src/Shared/Recruit.Vacancies.Client/Infrastructure/QueryStore/Projections/Employer/EmployerDashboard.cs]: 🙈

File: src/Employer/Employer.Web/Controllers/DashboardController.cs (hunk 1 above)
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Controllers/DashboardController.cs]: 👍

File: src/Employer/Employer.Web/Controllers/DashboardController.cs (hunk 2 above)
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Controllers/DashboardController.cs]: 👍

File: src/Employer/Employer.Web/Orchestrators/DashboardOrchestrator.cs (hunk 3 above)
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Orchestrators/DashboardOrchestrator.cs]: I was thinking along the same lines.  I'm going to keep it as it is for now until I've implemented a few alerts and then see if its worth pulling out common behaviour.

File: src/Employer/Employer.Web/Views/VacancyPreview/VacancyPreview.cshtml (hunk 4 above)
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Views/VacancyPreview/VacancyPreview.cshtml]: Will keep in mind.

File: src/Shared/Recruit.Vacancies.Client/Infrastructure/QueryStore/Projections/Employer/EmployerDashboard.cs (hunk 5 above)
Comment (chrisjensenuk) [reply src/Shared/Recruit.Vacancies.Client/Infrastructure/QueryStore/Projections/Employer/EmployerDashboard.cs]: :+1:

**Comment by chrisjensenuk** [comment]
Changes were in #697

**Pull request closed without merging**